src/main/java/org/folio/circulation/domain/CreateRequestService.java

```diff
@@ -6,78 +6,84 @@
 
+import java.lang.invoke.MethodHandles;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Objects;
 import java.util.Optional;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
 
 import org.folio.circulation.domain.policy.RequestPolicy;
 import org.folio.circulation.domain.policy.RequestPolicyRepository;
 import org.folio.circulation.support.ResponseWritableResult;
 import org.folio.circulation.support.Result;
 import org.folio.circulation.support.http.server.ValidationError;
 
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+
 public class CreateRequestService {
+
+  private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
   private final RequestRepository requestRepository;
   private final UpdateItem updateItem;
   private final UpdateLoanActionHistory updateLoanActionHistory;
   private final RequestPolicyRepository requestPolicyRepository;
+  private final LoanRepository loanRepository;
   private final UpdateLoan updateLoan;
 
-  public CreateRequestService(
-    RequestRepository requestRepository,
-    UpdateItem updateItem,
-    UpdateLoanActionHistory updateLoanActionHistory,
-    UpdateLoan updateLoan,
-    RequestPolicyRepository requestPolicyRepository) {
+  public CreateRequestService(RequestRepository requestRepository, UpdateItem updateItem,
+      UpdateLoanActionHistory updateLoanActionHistory, UpdateLoan updateLoan,
+      RequestPolicyRepository requestPolicyRepository, LoanRepository loanRepository) {
 
     this.requestRepository = requestRepository;
     this.updateItem = updateItem;
     this.updateLoanActionHistory = updateLoanActionHistory;
     this.updateLoan = updateLoan;
     this.requestPolicyRepository = requestPolicyRepository;
+    this.loanRepository = loanRepository;
   }
 
   public CompletableFuture<Result<RequestAndRelatedRecords>> createRequest(
-    RequestAndRelatedRecords requestAndRelatedRecords) {
+      RequestAndRelatedRecords requestAndRelatedRecords) {
 
     return completedFuture(refuseWhenItemDoesNotExist(requestAndRelatedRecords)
-      .next(CreateRequestService::refuseWhenInvalidUserAndPatronGroup)
-      .next(CreateRequestService::refuseWhenItemIsNotValid)
-      .next(CreateRequestService::refuseWhenUserHasAlreadyRequestedItem))
-      .thenComposeAsync(r -> r.after(requestPolicyRepository::lookupRequestPolicy))
-      .thenApply(r -> r.next(CreateRequestService::refuseWhenRequestCannotBeFulfilled))
-      .thenApply(r -> r.map(CreateRequestService::setRequestQueuePosition))
-      .thenComposeAsync(r -> r.after(updateItem::onRequestCreation))
-      .thenComposeAsync(r -> r.after(updateLoanActionHistory::onRequestCreation))
-      .thenComposeAsync(r -> r.after(updateLoan::onRequestCreation))
-      .thenComposeAsync(r -> r.after(requestRepository::create));
+        .next(CreateRequestService::refuseWhenInvalidUserAndPatronGroup)
```

Comment:
Why have these been further indented? Are they using 4 spaces per indentation level?

Reply:
I believe they are in keeping with the indentation I found in place initially, but I could be wrong. I will double check over the formatting there.